### PR TITLE
Added TeamDrive v 4.2.1

### PR DESCRIPTION
**Pull request by franklouwers**
### Checklist
- [x] The commit message includes the cask’s name and version.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.

Additionally, when **adding a new cask**:
- [x] Checked there are no open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there are no closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.

### CODE REVIEW

File: Casks/teamdrive.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  license :closed
+
+  installer script: 'Install-TeamDrive-4.2.1.1547_TMDR.app/Contents/MacOS/osx-intel',
+            args:   ['--unattendedmodeui', 'none', '--mode', 'unattended']
+
+  uninstall script: {
+                      executable: '/Applications/TeamDrive/uninstall.app/Contents/MacOS/osx-intel',
+                      args:       ['--mode', 'unattended'],
+                    },
+            signal: [
```
Comment (franklouwers) [inline Casks/teamdrive.rb]: this is ugly, but the processes don't stop after running the uninstall script or sending regular term/quit signals

File: Casks/teamdrive.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+cask 'teamdrive' do
+  version '4.2.1'
```
Comment (vitorgalvao) [inline Casks/teamdrive.rb]: `version '4.2.1547'`

File: Casks/teamdrive.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  sha256 '9035faf2670bd77259667e69952a392f535be7ba116f8a291ee79e700e5099bd'
+
+  # s3download.teamdrive.net.s3.amazonaws.com/4.2.1547/TMDR/mac-10.10.5/Install-TeamDrive-4.2.1.1547_TMDR.dmg was verified as official when first introduced to the cask
+  url 'http://s3download.teamdrive.net.s3.amazonaws.com/4.2.1547/TMDR/mac-10.10.5/Install-TeamDrive-4.2.1.1547_TMDR.dmg'
```
Comment (vitorgalvao) [inline Casks/teamdrive.rb]: `url "http://s3download.teamdrive.net.s3.amazonaws.com/#{version}/TMDR/mac-10.10.5/Install-TeamDrive-#{version}_TMDR.dmg"`

File: Casks/teamdrive.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  name 'TeamDrive'
+  homepage 'https://www.teamdrive.com/'
+  license :closed
+
+  installer script: 'Install-TeamDrive-4.2.1.1547_TMDR.app/Contents/MacOS/osx-intel',
```
Comment (vitorgalvao) [inline Casks/teamdrive.rb]: `installer script: "Install-TeamDrive-#{version}_TMDR.app/Contents/MacOS/osx-intel",`

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by franklouwers** [comment]
changed to as per @vitorgalvao 's request.

File: Casks/teamdrive.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+cask 'teamdrive' do
+  version '4.2.1.1547'
+  sha256 '9035faf2670bd77259667e69952a392f535be7ba116f8a291ee79e700e5099bd'
+
+  # s3download.teamdrive.net.s3.amazonaws.com/4.2.1547/TMDR/mac-10.10.5/Install-TeamDrive-4.2.1.1547_TMDR.dmg was verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/teamdrive.rb]: This is also too detailed. As it is, it would need to be redone every time, which defeats the purpose of interpolation.

**Comment by vitorgalvao** [comment]
Thank you for the contribution. It needed some fixes, so they were made in 9f4fdf2166f390db44a56c74374e4cd4f3393a78. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

File: Casks/teamdrive.rb (hunk 5 above)
Comment (franklouwers) [inline Casks/teamdrive.rb]: "brew cask style" forced me to do that. If I just wrote the s3 hostname, it complained. I had to write the entire url to make cask style happy. Is that a bug?

Comment (vitorgalvao) [inline Casks/teamdrive.rb]: What did you write, exactly? Becuase [this](https://github.com/caskroom/homebrew-cask/blob/0de180ec6df10ad306a10c9494580998d8a2a8b7/Casks/teamdrive.rb#L5) passes fine.

**Pull request closed without merging**